### PR TITLE
Issue179 - fix sti hangs when using non-existent latest

### PR DESCRIPTION
**Pull request by soltysh**
This PR mainly addresses #179, but it also contains removals mentioned in #199.
@bparees @csrwng PTAL

### CODE REVIEW

File: pkg/docker/docker.go [hunk 1]
```diff
@@ -22,6 +23,7 @@ const (
 	LocationLabel   = "io.openshift.sti.location"
 
 	DefaultLocation = "/tmp"
+	DefaultTag      = ":latest"
```
Comment (mfojtik) [inline pkg/docker/docker.go]: remove ':'

File: pkg/docker/docker.go [hunk 2]
```diff
@@ -178,6 +180,10 @@ func (d *stiDocker) CheckAndPull(name string) (image *docker.Image, err error) {
 
 // PullImage pulls an image into the local registry
 func (d *stiDocker) PullImage(name string) (image *docker.Image, err error) {
+	_, tag := parsers.ParseRepositoryTag(name)
+	if len(tag) == 0 {
+		name += DefaultTag
```
Comment (mfojtik) [inline pkg/docker/docker.go]: use strings.Join()

**Comment by csrwng** [comment]
Should we also not do the same logic for running a container?

**Comment by soltysh** [comment]
@mfojtik addressed your comments. @csrwng you're right, I've added that to all our methods where I though it'd make sense. PTAL again.

**Comment by csrwng** [comment]
LGTM

**Comment by soltysh** [comment]
[merge]

**Pull request merged**